### DISCUSSION TITLE
kcadm.sh credentials login failed when the default browser flow's OTP is REQUIRED!

### DISCUSSION THREAD

**Discussion by ilu123** [keycloak/keycloak#21472] [Archive]
Hi Team,
RT.
When we setup the _built-in_ **browser** authentication flow's **OTP** to be **REQUIRED**, then the `kcadm.sh config credentials --server http://127.0.0.1:8080/ --realm master --user $KEYCLOAK_ADMIN --password $KEYCLOAK_ADMIN_PASSWORD`
will then be failed as of  _INVALIDE_CREDENTIAL_.

Is there anyway to resolve this with OTP required enabled?

Thanks in advance.

**Comment by adrianlanzi** [comment]
Any update on this issue?

**Reply by codespearhead** [reply]
Are you still facing this issue @ilu123 and @adrianlanzi ?

If so, show me the commands you ran, including the `bin/kc.sh start-dev`, so I can reproduce it in my machine.

Also, tell us what Keycloak version you are running and show us what is logged to the terminal running Keycloak when you reproduce the error on your end.

**Comment by ilu123** [comment]
Hi sorry for the late reply.
I figured out the reason: It's caused by the flow's authentication required option. 
The final solution I found out is to add one condition for the admin user to login without OTP code.

**Reply by adrianlanzi** [reply]
Thanks @ilu123 ! I'll try it
